### PR TITLE
Makes shift+return create a new line in a text tile

### PR DESCRIPTION
**Pull request by rodfersou**
- [x] Add shift+return for Text tile
- [x] Added cypress tests for every tile
image

Those should be fixed on other PR:

- The hero tile needs to implement react dropzone to be able to finish the test. https://github.com/plone/volto/issues/804
- Table tile not working (at least for tests). https://github.com/plone/volto/issues/805
- Tiles Image, Video and Maps don't work in Guilhotina. https://github.com/plone/volto/issues/806
- Improve Text tile test. https://github.com/plone/volto/issues/807

### CODE REVIEW

**Comment by tisto** [comment]
@rodfersou the new cypress test fails.

**Comment by robgietema** [comment]
The description is saved as a string right? and rendered HTML, so the newline will not be visible. What is the reason we want this functionality?

**Comment by tisto** [comment]
@robgietema @rodfersou shift-return should work in regular text tiles to being able to create quotes where you control the carriage return. No idea why that ended up in the description field.

**Comment by rodfersou** [comment]
@tisto @robgietema should be green after last change.

Please take a look at this comment where I sum up the state of the PR and tests https://github.com/plone/volto/pull/785#issue-290806398

**Comment by rodfersou** [comment]
finally green!

File: cypress/integration/tiles.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
+describe('Default Tiles functionality', () => {
```
Comment (tisto) [inline cypress/integration/tiles.js]: There is no "default" tile. This is the "Text tile".

File: cypress/integration/tiles.js [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+    // View
+    if (Cypress.env('API') === 'plone') {
+      cy.get('#page-document h3').should('have.text', expected);
+    } else {
+      // guillotina
+      cy.contains(expected);
+    }
+  });
+
+  it('Table Tile', () => {
```
Comment (tisto) [inline cypress/integration/tiles.js]: There is no table tile in master yet. Therefore we should remove this test altogether.

File: src/components/manage/Tiles/Text/Edit.jsx [hunk 3]
```diff
@@ -302,7 +307,12 @@ export default class Edit extends Component {
               <React.Fragment>
                 <div className="separator" />
                 <Button.Group>
-                  <Button icon basic onClick={this.openCustomTileMenu}>
+                  <Button
+                    icon
+                    basic
+                    className="show-hidden-tiles"
```
Comment (tisto) [inline src/components/manage/Tiles/Text/Edit.jsx]: @rodfersou is "show-hidden-tiles" an existing class in our css?

File: cypress/integration/tiles.js (hunk 2 above)
Comment (robgietema) [reply cypress/integration/tiles.js]: Yes there is.

File: src/components/manage/Tiles/Text/Edit.jsx (hunk 3 above)
Comment (rodfersou) [reply src/components/manage/Tiles/Text/Edit.jsx]: I just use this class to fire the button in the test

File: cypress/integration/tiles.js (hunk 1 above)
Comment (rodfersou) [reply cypress/integration/tiles.js]: maybe we can review the structure of the tests later, for something that makes more sense.
right now I just put all tests  in the same `describe`

File: cypress/integration/tiles.js (hunk 2 above)
Comment (tisto) [reply cypress/integration/tiles.js]: @robgietema w00t. That's fantastic! Man, do you ever eat or sleep like other humans? ;)

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,7 @@
 
 ### Added
 
+- Makes shift+return create new line in Description field @rodfersou
```
Comment (tisto) [inline CHANGELOG.md]: @rodfersou pls replace "Description field" with "Text tile".

Comment (rodfersou) [reply CHANGELOG.md]: done

**Comment by tisto** [comment]
@rodfersou please try not to mix two different issues (shift+return, tiles tests) into one PR in the future. This makes things harder to review and merge.

**Comment by sneridagh** [comment]
@tisto FYI I've fixed a lot of issues and created a special redraftJS `toHTML` function to deal with this, I don't think that it's backported yet.

**Comment by tisto** [comment]
@rodfersou could you image to look into this again after Victor's refactoring? Maybe it is better to create a new branch for this.

**Comment by tisto** [comment]
@rodfersou it turns out we had to put quite some effort into make this work in our recent project. We have the code in one of our projects and we need to copy it over to Volto core.

**Comment by sneridagh** [comment]
Superseeded by #1044. @rodfersou I tried the ctrl+shift test but it failed :(
It would be great if we could extract this test and the other tests for the blocks that you did with this PR and update them and add it to the build. Not urgent though!

**Comment by tisto** [comment]
Fixed in master. Closing.

**Pull request closed without merging**